src/parser/jobs/drk/modules/Delirium.tsx

```diff
@@ -39,14 +41,92 @@ export class Delirium extends BuffWindow {
 				{action: this.data.actions.SCARLET_DELIRIUM, expectedPerWindow: 1},
 				{action: this.data.actions.COMEUPPANCE, expectedPerWindow: 1},
 				{action: this.data.actions.TORCLEAVER, expectedPerWindow: 1},
+				{action: this.data.actions.IMPALEMENT, expectedPerWindow: 3},
 			],
 			suggestionIcon: this.data.actions.DELIRIUM.icon,
 			suggestionContent: <Trans id="drk.delirium.suggestions.gcdactions.content">
-				Each <DataLink action="DELIRIUM" /> window should contain <DataLink action="SCARLET_DELIRIUM" />, <DataLink action="COMEUPPANCE" />, and <DataLink action="TORCLEAVER" />.
-				Using regular weaponskills resets your Delirium combo progress and causes you to lose the increased potency of the comboed skills.
+				Each <DataLink action="DELIRIUM" /> window should contain <DataLink action="SCARLET_DELIRIUM" />, <DataLink action="COMEUPPANCE" />, and <DataLink action="TORCLEAVER" />, or three <DataLink action="IMPALEMENT" />s.
+				Using non-Delirium combo actions resets your Delirium combo progress and causes you to lose the increased potency of the comboed skills.
 			</Trans>,
 			suggestionWindowName,
 			severityTiers: SEVERITIES.WRONG_GCDS,
+			adjustCount: this.adjustCount.bind(this),
+			adjustOutcome: this.adjustOutcome.bind(this),
 		}))
 	}
+
+	private threeImpalementsUsedInWindow(window: HistoryEntry<EvaluatedAction[]>): boolean {
+		const totalNumberOfDeliriumStacks = 3
+		const threeImpalementsUsed = window.data.filter(event => (event.action.id === this.data.actions.IMPALEMENT.id)).length === totalNumberOfDeliriumStacks
+		return threeImpalementsUsed
+	}
+
+	private allThreeComboActionsUsed(window: HistoryEntry<EvaluatedAction[]>): boolean {
+		const scarletUsed = window.data.filter(event => (event.action.id === this.data.actions.SCARLET_DELIRIUM.id)).length > 0
+		const comeuppanceUsed = window.data.filter(event => (event.action.id === this.data.actions.COMEUPPANCE.id)).length > 0
+		const torcleaverUsed = window.data.filter(event => (event.action.id === this.data.actions.TORCLEAVER.id)).length > 0
+		return scarletUsed && comeuppanceUsed && torcleaverUsed
+	}
+
+	// We want to allow the following mixtures of AoE/single target:
+	// - 2x Impalement, 1x Scarlet Delirium
+	// - 1x Scarlet Delirium, 1x Comeuppance, 1x Impalement
+	// Anything else (e.g. 2x Scarlet Delirium, 1x Impalement) is disallowed
+	private deliriumMixtureUsed(window: HistoryEntry<EvaluatedAction[]>): boolean {
+		const scarletUsed = window.data.filter(event => (event.action.id === this.data.actions.SCARLET_DELIRIUM.id)).length
+		const comeuppanceUsed = window.data.filter(event => (event.action.id === this.data.actions.COMEUPPANCE.id)).length
+		const impalementUsed = window.data.filter(event => (event.action.id === this.data.actions.IMPALEMENT.id)).length
+		return (scarletUsed === 2 && impalementUsed === 1) || (scarletUsed === 1 && comeuppanceUsed === 1 && impalementUsed === 1)
```

Comment:
```suggestion
		return (scarletUsed === 1 && impalementUsed === 2) || (scarletUsed === 1 && comeuppanceUsed === 1 && impalementUsed === 1)
```

Checked values backwards for 2 impalement + 1 scarlet case.

Reply:
I'm dumb, thank you